authorizerd.go

```diff
@@ -13,34 +13,38 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-package providerd
+package authorizerd
 
 import (
 	"context"
+	"crypto/x509"
+	"fmt"
 	"net/http"
+	"strings"
 	"time"
 
 	"github.com/kpango/gache"
 	"github.com/kpango/glg"
 
 	"github.com/pkg/errors"
-	"github.com/yahoojapan/athenz-policy-updater/policy"
-	"github.com/yahoojapan/athenz-policy-updater/pubkey"
-	"github.com/yahoojapan/athenz-policy-updater/role"
+	"github.com/yahoojapan/athenz-authorizer/policy"
+	"github.com/yahoojapan/athenz-authorizer/pubkey"
+	"github.com/yahoojapan/athenz-authorizer/role"
 )
 
-// Providerd represents a daemon for user to verify the role token
-type Providerd interface {
-	StartProviderd(context.Context) <-chan error
+// Authorizerd represents a daemon for user to verify the role token
+type Authorizerd interface {
+	Start(ctx context.Context) <-chan error
 	VerifyRoleToken(ctx context.Context, tok, act, res string) error
-	//	VerifyRoleCert(cert []*x509.Certificate) error
+	VerifyRoleJWT(ctx context.Context, tok, act, res string) error
+	VerifyRoleCert(ctx context.Context, peerCerts []*x509.Certificate, act, res string) error
 }
 
-type provider struct {
+type authorizer struct {
 	//
-	pubkeyd         pubkey.Pubkeyd
-	policyd         policy.Policyd
-	roleTokenParser role.RoleTokenParser
+	pubkeyd    pubkey.Pubkeyd
+	policyd    policy.Policyd
+	roleParser role.Role
```

Comment:
the package name `role` seems not accurate, and the variable name should be roleTokenParser or rtParser.
```suggestion
	rtParser roletok.Parser
```

Reply:
I think Role Interface is not just Parsing it also verify credential

Reply:
maybe something like `roleTokenProcessor` is better?

Reply:
fixed!